packages/app/src/debug/DebugRunNavigation.vue

```diff
@@ -68,56 +68,63 @@
+            data-cy="debug-historical-runs"
           >
-            <div class="flex my-10px mx-16px items-center">
-              <DebugCommitIcon class="flex-shrink-0" />
-              <LightText class="flex-shrink-0 ml-12px truncate">
-                {{ sha.slice(0, 7) }}
-              </LightText>
-              <Dot />
-              <span
-                class="font-medium text-sm text-gray-800 truncate"
-                :title="groupByCommit[sha].message!"
-              >
-                {{ groupByCommit[sha].message }}
-              </span>
-              <span
-                v-if="sha === currentCommitInfo?.sha"
-                data-cy="tag-checked-out"
-                class="border rounded font-medium border-gray-100 border-1 flex-shrink-0
+            <li
+              v-for="sha of Object.keys(groupByCommit)"
+              :key="sha"
+              :data-cy="`commit-${sha}`"
+            >
+              <div class="flex my-10px mx-16px items-center">
+                <DebugCommitIcon class="flex-shrink-0" />
+                <LightText class="flex-shrink-0 ml-12px truncate">
+                  {{ sha.slice(0, 7) }}
+                </LightText>
+                <Dot />
+                <span
+                  class="font-medium text-sm text-gray-800 truncate"
+                  :title="groupByCommit[sha].message!"
+                >
+                  {{ groupByCommit[sha].message }}
+                </span>
+                <span
+                  v-if="sha === currentCommitInfo?.sha"
+                  data-cy="tag-checked-out"
+                  class="border rounded font-medium border-gray-100 border-1 flex-shrink-0
               h-16px ml-8px px-4px text-12px text-purple-400 leading-16px
               align-middle inline-flex items-center"
-              >
-                Checked out
-              </span>
-            </div>
-
-            <ul v-if="groupByCommit[sha].runs">
-              <DebugRunNavigationRow
-                v-for="run of groupByCommit[sha].runs"
-                :key="run?.runNumber!"
-                :run-number="run?.runNumber!"
-                :is-current-run="isCurrentRun(run!)"
-                :gql="run!"
-                @change-run="changeRun(run!)"
-              />
-            </ul>
-          </li>
-        </ul>
+                >
+                  Checked out
+                </span>
+              </div>
+
+              <ul v-if="groupByCommit[sha].runs">
+                <DebugRunNavigationRow
+                  v-for="run of groupByCommit[sha].runs"
+                  :key="run?.runNumber!"
+                  :run-number="run?.runNumber!"
+                  :is-current-run="isCurrentRun(run!)"
+                  :gql="run!"
+                  @change-run="changeRun(run!)"
+                />
+              </ul>
+            </li>
+          </ul>
+        </div>
+        <div
+          v-if="runs.length === 100"
```

Comment:
We limit this query to 100 runs, so if we have exactly 100 runs, then we're either at or over the limit, so display the message.